src/nimtenbrew/ctr/icon.nim

```diff
@@ -0,0 +1,83 @@
+import ../strings
+
+import strutils
+import strformat
+
+import nimPNG
+import stew/endians2
+
+proc load*(png_path: string, size: int): string =
+    ## Loads a PNG icon at `png_path` with expected size `size`
+    ## Returns the data buffer loaded by nimPNG
+
+    try:
+        let png = nimPNG.loadPNG32(png_path)
```

Comment:
I'm personally a fan of Pixie (https://github.com/treeform/pixie) for loading images, but don't really care which library you use. Pixie has JPEG support too, which is nice.

Reply:
Only reason I'd chosen nimPNG was I do want to keep the library small. Although I'm not familiar with building a library in nim as much as you, so maybe the size is not much to worry about.